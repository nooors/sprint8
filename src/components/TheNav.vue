<template>
  <nav>
    <div class="nav">
      <router-link :to="{ name: 'Home' }">HOME</router-link
      ><router-link :to="{ name: 'Starships' }">STARSHIPS</router-link>
    </div>
  </nav>
</template>

<script>
export default {
  name: "TheNav",
};
</script>

<style lang="css" scoped>
.nav {
  width: 100%;
  background-color: black;
  display: flex;
  justify-content: center;
  border-bottom: 1px solid rgb(134, 133, 133);
  border-top: 1px solid rgb(134, 133, 133);
}

.nav a:focus {
  border-bottom: 3px solid blue;
}

.nav a:last-child {
  border-right: 1px solid rgb(134, 133, 133);
}

.nav a {
  padding: 0.5rem 1rem;
  border-left: 1px solid rgb(134, 133, 133);
  position: relative;
}
</style>
