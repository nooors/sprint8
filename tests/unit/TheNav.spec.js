import { shallowMount, mount } from "@vue/test-utils";
import TheNav from "@/components/TheNav.vue";

describe("TheNav.vue", () => {
  it("Renders the component", () => {
    const wrapper = shallowMount(TheNav, {
      stubs: ["router-link", "router-view"],
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});

test("Render apropiate text", () => {
  const wrapper = mount(TheNav, {
    stubs: ["router-view", "router-link"],
  });
  expect(wrapper.text()).toBe("HOMESTARSHIPS");
});
