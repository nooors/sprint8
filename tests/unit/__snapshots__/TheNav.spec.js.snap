// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheNav.vue Renders the component 1`] = `
<nav>
  <div class="nav">
    <router-link-stub to="[object Object]">HOME</router-link-stub>
    <router-link-stub to="[object Object]">STARSHIPS</router-link-stub>
  </div>
</nav>
`;
